### DISCUSSION TITLE
Password protecting workbooks

### DISCUSSION THREAD

**Discussion by simonjcarr** [exceljs/exceljs#2524] [Ideas & Proposals]
Is there any functionality available in exceljs to protect / unprotect a workbook. If not how do other developers using exceljs get around this?

**Comment by junaidinam** [comment]
One way to get around this is to zip your excel file and password protect that instead.

**Comment by simonjcarr** [comment]
@Siemienik I am not looking to prevent people for seeing my workbooks. I want to send a workbook to a supplier with everything locked other than the cells I want them to fill in and then send back to me.

**Comment by r4tz52** [comment]
As far as I see the "password protection feature" is not rocket science: https://github.com/dtjohnson/xlsx-populate/blob/master/lib/Encryptor.js
UPD: moreover, it can be done externally - the "encrypt" function takes Buffer with XLSX contnet as argument and returns Buffer with encrypted XLSX file.

**Comment by helloimthinh** [comment]
that's also a question that i have, waiting waiting ...

**Comment by steresi** [comment]
It appears that `exceljs` is not able to decrypt a password-protected Excel workbook (password is prompted at time the Excel file is opened). I hope it may be added, as the `xlsx-populate` package hasn't been updated in over two years! https://www.npmjs.com/package/xlsx-populate

**Comment by ShadabFaiz** [comment]
Isn't this functionality already implemented ? I can protect any cell from editing and allow only certain cells to be editable. However this does not work if you upload the file on google drive and open using google sheet

**Reply by kunalindoliya** [reply]
got any solution for google sheet ?

**Comment by sanjusabu** [comment]
Is this done now, or is it not available?
